javascript/ql/src/semmle/javascript/dataflow/DefinedCustomAbstractValues.qll

```diff
@@ -0,0 +1,153 @@
+/**
+ * Provides classes for working with analysis-specific abstract values.
+ *
+ * Implement a subclass of `CustomAbstractValueDefinition` When the builtin
+ * abstract values of `AbstractValues.qll` are not expressive enough.
+ *
+ * For performance reasons, all subclasses of `CustomAbstractValueDefinition`
+ * should be part of the standard library.
+ */
+
+import javascript
+private import internal.AbstractValuesImpl
+private import InferredTypes
+
+/**
+ * An abstract representation of an analysis-specific value.
+ *
+ * Wraps a `CustomAbstractValueDefinition`.
+ */
+class DefinedCustomAbstractValue extends AbstractValue, TDefinedCustomAbstractValue {
```

Comment:
I don't like the "defined" bit. How about `CustomAbstractValueFromDefinition`? It's a fairly awkward name, but I expect developers don't really need to deal with it directly (presumably they would mostly use `CustomAbstractValueDefinition.getAbstractValue` to get the abstract value induced by a definition).